tools/CMakeLists.txt

```diff
@@ -0,0 +1,22 @@
+add_executable (mraa-gpio mraa-gpio.c)
+add_executable (mraa-i2c mraa-i2c.c)
+add_executable (mraa-uart mraa-uart.c)
+
+include_directories(${PROJECT_SOURCE_DIR}/api)
+# FIXME Hack to access mraa internal types used by mraa-i2c
+include_directories(${PROJECT_SOURCE_DIR}/include)
+include_directories(${PROJECT_SOURCE_DIR}/api/mraa)
+
+target_link_libraries (mraa-gpio mraa)
+target_link_libraries (mraa-i2c mraa)
+target_link_libraries (mraa-uart mraa)
+
+if (INSTALLGPIOTOOL AND NOT INSTALLTOOLS)
```

Comment:
just remove INSTALLGPIOTOOLS and make INSTALLTOOLS just TOOLS or something that enables the building and installating of them. This was kind of a mess when we started adding these and install logic was never really corrected since these where kinda half sample/half utils.

Reply:
makes sense! I was tempted to remove that initially but fell back

Reply:
+1

Reply:
+1. Nice work. Waiting for someone to write mraa-spi.

Reply:
I can write it but don't have any supported board to test it. Sadly SPI is not exposed in all of our 96Boards